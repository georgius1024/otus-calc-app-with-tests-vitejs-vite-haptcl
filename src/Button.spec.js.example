import { describe, it, expect, beforeAll, beforeEach, vi } from 'vitest';
import { mount } from '@vue/test-utils';
import component from './Button.vue';

describe('Button', () => {
  const buildWrapper = (char) => {
    const propsData = {
      char,
    };
    return mount(component, {
      propsData,
    });
  };
  it('mounts without error', () => {
    const wrapper = buildWrapper('*');
    expect(wrapper.exists).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('emits when button clicked', async () => {
    const wrapper = buildWrapper('*');
    await wrapper.find('[data-testid="btn-*"]').trigger('click');
    expect(wrapper.emitted().click).toHaveLength(1);
  });
});
