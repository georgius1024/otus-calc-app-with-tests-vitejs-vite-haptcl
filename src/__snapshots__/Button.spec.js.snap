// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > mounts without error 1`] = `"<button data-testid="btn-*">*</button>"`;
